doc/glossary.rst

```diff
@@ -0,0 +1,116 @@
+        A :term:`JSON` configuration file defining :term:`Framework` :term:`engine` behaviour -
+        which plugins and widgets to use. Is validated against a :term:`schema`.
+        Example of a definition is the *Maya Geometry* publisher. Definitions lives within
+        the ftrack-connect-pipeline-definition plugin, source code can be found here:
+        https://github.com/ftrackhq/ftrack-connect-pipeline-definition.git
+
+    Engine
+        A core Python module driving a specific behaviour within the :term:`Framework`,
+        for example publishing or asset management.
+
+    Event manager
+        A module responsible sendimg and receiving ftrack events, through the
+        :term:`ftrack Python API`.
+
+    Framework
+        A framework is a structure that you can build software on. It serves as a foundation,
+        so you're not starting entirely from scratch. Frameworks are typically associated
+        with a specific programming language and are suited to different types of tasks.
+        The ftrack pipeline Framework is a set of modules/layers enabling asset publish,
+        load, management and other core functionality within an :term:`DCC` application
+        or standalone. The core framework module is called ftrack-connect-pipeline, with
+        source code to be found here: https://github.com/ftrackhq/ftrack-connect-pipeline.git
+
+    Host
+        The central part of the core framework that discovers and executes definitions
+        through engines, handle the context and much more. The host is designed to
+        be able to operate in remote mode through the ftrack event system.
+
+    Host type
+        The host type is the actual DCC type and is used to identify a DCC module
+        and bind a definition to the DCC application. An example host types value
+        is **maya**.
+
+    JSON
+        JSON is a lightweight format for storing and transporting data, and stands
+        for JavaScript Object Notation. For more information
+        https://www.json.org/
+
+    Plugin
+        A module designed to be discovered by the :term:`ftrack Python API`. Plugins
+        designed to be discovered by Connect is called Connect plugins and are main
+        components of the :term:`Framework`. Framework plugins resides within the
+        definition module and are referenced from the with the :term:`definition` JSON
+        configurations.
+
+    Python
+        A programming language that lets you work more quickly and integrate
+        your systems more effectively. Often used in creative industries. Visit
+        the language website at http://www.python.org
+
+    Qt
+        The default UI framework utilised by the Framework, through PySide and
+        the Qt.py Python binding module. The corresonding Framework module containing
+        UI bindings is named ftrack-connect-pipeline-qt and its source code can be
+        found here: https://github.com/ftrackhq/ftrack-connect-pipeline-qt.git
```

Comment:
@henriknorin-ftrack no need to reference the repo here.
One thing is the glossary for the technology ,another is how we use it.